plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java

```diff
@@ -1642,8 +1646,37 @@ public void redrawData(boolean attributesChanged, boolean rowsChanged)
         //this.updateStatusMessage();
     }
 
+    private final UIJob statusBarLayoutJob = new UIJob("Pending resultset view status bar relayout") {
```

Comment:
Why is layout performed in a job anyways? Can't it be run using `UIUtils.asyncExec` instead?

IIRC you don't even have to run it on the EDT thread since you're inside an event handler for a UI event.

Reply:
Because the job scheduling mechanism guarantees that the job would be executed only once after all the already running UI activities would be finished. It is possible to have a sequence of resize events here as a result of settings change for multiple toolbar contributions at once. So we need only one relayout then, after the configuration update would be applied. The use of asyncExec leads to an unnecessary sequence of relayouts giving an unappropriate result according to my tests.